### PR TITLE
[dagster-fivetran] Implement sync_and_poll method in FivetranClient

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

This PR uses the sync and poll methods implemented in previous PRs to implement `sync_and_poll` in `FivetranClient`. This method will be used in a subsequent PR to materialize Fivetran assets.

Tests are added to test the full sync and poll behavior.

## How I Tested These Changes

Additional unit tests with BK

### CODE REVIEW

**Comment by maximearmstrong** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26061?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26062** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26062?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26061** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26061?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26061?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26060** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26060?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26059** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26059?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25911** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25911?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#25944](https://github.com/dagster-io/dagster/pull/25944) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25944?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#25889** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25889?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25808** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25808?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25807** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25807?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25797** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25797?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25788** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25788?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25756** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25756?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25751** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25751?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25750** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25750?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py [hunk 1]
```diff
@@ -108,3 +108,72 @@ def test_basic_resource_request(
             poll_timeout=2,
             poll_interval=1,
         )
+
+
+@pytest.mark.parametrize(
+    "n_polls, succeed_at_end",
+    [(0, True), (0, False), (4, True), (4, False), (30, True)],
+)
```
Comment (dpeng817) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py:116]: nit: add test ids

File: python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py [hunk 2]
```diff
@@ -108,3 +108,72 @@ def test_basic_resource_request(
...
+    )
+    client = resource.get_client()
+
+    test_connector_api_url = get_fivetran_connector_api_url(connector_id)
+
+    test_succeeded_at = TEST_MAX_TIME_STR
+    test_failed_at = TEST_PREVIOUS_MAX_TIME_STR
+    # Set `failed_at` as more recent that `succeeded_at` if the sync and poll process is expected to fail
+    if not succeed_at_end:
+        test_succeeded_at, test_failed_at = test_failed_at, test_succeeded_at
```
Comment (dpeng817) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py:129]: this is kind of a weird way to do this but I think it's fine. Just add a comment

**Review by dpeng817 — APPROVED** [review]

**Comment by dpeng817** [comment]
Nice, thanks

File: python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py (hunk 2 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py]: I changed it in [05442c2](https://github.com/dagster-io/dagster/pull/26061/commits/05442c2eb793989ba4d8792280489c42c4272da6) to make it clearer

File: python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py (hunk 1 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_resources.py:117]: Done in [e023cd6](https://github.com/dagster-io/dagster/pull/26061/commits/e023cd6b82353f3f51d4ebb884335b7d37252871)

**Pull request merged**